### DISCUSSION TITLE
Overview of Build Process / GitHub Actions

### DISCUSSION THREAD

**Discussion by mariusfilipowski** [actions/virtual-environments#7397] [General questions]
I am trying to understand how Ubuntu images are built with this great repo. 
I checked the Workflows and Build Runs, but I am still puzzled a little. Is there an overview of the build and release process for this repo?
Some questions I have:
- I see that there are repository dispatch workflows. But I cannot see who triggers them and when? What are the parameter values?
- The Ubuntu Images seem to have no successful run in the past and seem to be waiting for a runner, but then they time out.
- What is the purpose of the SBOM generated? What is it used for exactly?
- When and why is a Release flagged as `prerelease`? And when is it updated to a real release?

I hope I can learn more from this great repo.
Thank you for your help.

**Comment by erik-bershel** [comment]
Hey @mariusfilipowski! We glade that you are interested in project we maintain. I'll try to answer questions of yours. Information about how builds and releases exactly happening not really so available - it's true. And I cannot share all the details due to a security reasons. But.
- @mikhailkoliada may better answer your first question than me
- Described situation looks like rare example of incident with Actions availability or overusing of self-hosted agents. Please provide a link to the failed run to clarify a situation.
-  In the context of GitHub Actions hosted runners, an SBOM details the software pre-installed on the virtual machine that is running your Actions workflows. This is useful in the situation where there is a vulnerability detected, you will be able to quickly tell if you are affected or not. If you are building artifacts, you can include this SBOM in your bill of materials for a comprehensive list of everything that went into creating your software.
- Our CD process takes at least three days - it is the main reason of using that flag. A new release is created with the prerelease flag. Upon completion of the delivery process, the flag is removed.

**Comment by mariusfilipowski** [comment]
Thanks a lot for the quick answer and for the information you can share. Appreciated!
Here the link: https://github.com/actions/runner-images/actions/workflows/ubuntu2004.yml 
I see here only failed or timed out builds of ubuntu.
 
But in the meantime I found out that the real build are running in Azure Devops here: https://github.visualstudio.com/runner-images/_build?definitionId=71&_a=summary

Thanks a lot. And I hope that @mikhailkoliada  can add more details.

Keep up the good work.

**Reply by erik-bershel** [reply]
Everything wright. Workflow you mentioned is under construction. :)
